lib/api/api.js

```diff
@@ -82,20 +84,50 @@ const api = {
                     }
                 }
             }
-            if (apiMethod === 'bucketPut') {
-                return bucketPut(userInfo, request, locationConstraint,
-                    log, callback);
-            }
-            if (apiMethod === 'objectCopy' ||
-                apiMethod === 'objectPutCopyPart') {
-                return this[apiMethod](userInfo, request, sourceBucket,
-                    sourceObject, log, callback);
-            }
+            // issue 100 Continue to the client
+            writeContinue(request, response);
             if (apiMethod === 'objectPut' || apiMethod === 'objectPutPart') {
                 return this[apiMethod](userInfo, request, streamingV4Params,
                     log, callback);
             }
-            return this[apiMethod](userInfo, request, log, callback);
+            const MAX_POST_LENGTH = request.method.toUpperCase() === 'POST' ?
+                1024 * 1024 : 1024 * 1024 / 2; // 1 MB or 512 KB
+            const post = [];
+            let postLength = 0;
+            request.on('data', chunk => {
+                postLength += chunk.length;
+                // Sanity check on post length
+                if (postLength <= MAX_POST_LENGTH) {
+                    post.push(chunk);
+                }
+                return undefined;
+            });
+
+            request.on('error', err => {
+                log.trace('error receiving request', {
+                    error: err,
+                });
+                return undefined;
```

Comment:
I think we want to call the callback back to the routes here with an internal error.